pkg/recipes/terraform/config/config.go

```diff
@@ -190,23 +237,26 @@ func getProviderConfigs(ctx context.Context, requiredProviders []string, ucpConf
 			return nil, err
 		}
 		if len(config) > 0 {
-			providerConfigs[provider] = config
+			providerConfigs[providerName] = config
 		}
 	}
 
 	return providerConfigs, nil
 }
 
-// AddTerraformBackend adds backend configurations to store Terraform state file for the deployment.
+// AddTerraformInfrastructure adds backend configurations to store Terraform state file for the deployment.
+// It also sets the required providers for the Terraform configuration.
 // Save() must be called to save the generated backend config.
 // Currently, the supported backend for Terraform Recipes is Kubernetes secret. https://developer.hashicorp.com/terraform/language/settings/backends/kubernetes
-func (cfg *TerraformConfig) AddTerraformBackend(resourceRecipe *recipes.ResourceMetadata, backend backends.Backend) (map[string]any, error) {
+func (cfg *TerraformConfig) AddTerraformInfrastructure(resourceRecipe *recipes.ResourceMetadata, backend backends.Backend, requiredProviders map[string]*RequiredProviderInfo) (map[string]any, error) {
```

Comment:
Not sure if `Infrastructure` is the right term to use here

Reply:
yes, open to suggestions, AddTerraformBackendAndReqProviders is another option.

Reply:
reverted changes to this function as discussed.